From ec156208692c5b83ab5cfc882e84da4991e973bb Mon Sep 17 00:00:00 2001
From: Minecrell <minecrell@minecrell.net>
Date: Fri, 29 Nov 2019 19:33:22 +0100
Subject: [PATCH 068/210] MSM8916: iommu/qcom: Select IOMMU_DMA instead of
 ARM_DMA_USE_IOMMU (v3)

The ARM32 IOMMU DMA implementation does not really work properly
with qcom_iommu. It fails to attach to any IOMMU because a domain
is created by default in the driver.

Now that there is IOMMU_DMA support for ARM32, select it instead
of ARM_DMA_USE_IOMMU.

v2: 'depend on' instead of 'select' to fix circular dependency
v3: Move "select IOMMU_DMA" to CONFIG_ARCH_MSM8916
---
 arch/arm/mach-qcom/Kconfig | 1 +
 drivers/iommu/Kconfig      | 1 -
 2 files changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-qcom/Kconfig b/arch/arm/mach-qcom/Kconfig
index 466acc4a5e0..92ec4a5c7f8 100644
--- a/arch/arm/mach-qcom/Kconfig
+++ b/arch/arm/mach-qcom/Kconfig
@@ -24,6 +24,7 @@ config ARCH_MSM8X60
 config ARCH_MSM8916
 	bool "Enable support for MSM8916"
 	select HAVE_ARM_ARCH_TIMER
+	select IOMMU_DMA
 	help
 	  Enable support for the Qualcomm Snapdragon 410 (MSM8916/APQ8016).
 
diff --git a/drivers/iommu/Kconfig b/drivers/iommu/Kconfig
index 3eb68fa1b8c..d5f878c3575 100644
--- a/drivers/iommu/Kconfig
+++ b/drivers/iommu/Kconfig
@@ -448,7 +448,6 @@ config QCOM_IOMMU
 	select QCOM_SCM
 	select IOMMU_API
 	select IOMMU_IO_PGTABLE_LPAE
-	select ARM_DMA_USE_IOMMU
 	help
 	  Support for IOMMU on certain Qualcomm SoCs.
 
-- 
2.34.1

